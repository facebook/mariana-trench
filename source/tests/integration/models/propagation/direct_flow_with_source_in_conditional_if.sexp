(method (private) "LFlow;.flow:(LSink;LSource;)V"
 (
  LINE(1)
  (const v0 0)
  (const v1 0)

  LINE(2)
  (if-eqz v0 :else)
  (invoke-virtual (v3) "LSource;.source:()LData;")
  (move-result-object v5)
  (goto :out)

  LINE(3)
  (:else)
  (new-instance "LData;")
  (move-result-object v5)

  LINE(4)
  (invoke-direct (v2) "LData;.<init>:()V")

  LINE(5)
  (:out)
  (invoke-virtual (v4 v5) "LSink;.sink:(LData;)V")

  (return-void)
 )
)
