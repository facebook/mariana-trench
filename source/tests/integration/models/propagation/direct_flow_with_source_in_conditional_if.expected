{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "issues" :
      [
        {
          "position" :
          {
            "line" : 5,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 5,
                "path" : "Flow.java"
              },
              "callee" : "LSink;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 1,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "call_position" :
              {
                "line" : 2,
                "path" : "Flow.java"
              },
              "callee" : "LSource;.source:()LData;",
              "callee_port" : "Return",
              "distance" : 1,
              "kind" : "TestSource",
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.flow:(LSink;LSource;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    }
  ]
}
